storage/src/adb/mod.rs

```diff
@@ -0,0 +1,18 @@
+//! Authenticated database (ADS) implementations.
```

Comment:
<img width="1005" alt="image" src="https://github.com/user-attachments/assets/5cdc9ff7-641a-498a-aa7c-bf7e31a07e94" />

For consistency, let's do `A mutable key-value database with provable state.` or `A collection of Authenticated Databases (ADB).`

Reply:
I think it's important to define ADB here so went with the second.